packages/block-editor/src/hooks/use-bindings-attributes.js

```diff
@@ -115,35 +121,26 @@ const BindingConnector = ( {
 		[ attrName, onPropValueChange ]
 	);
 
-	useEffect( () => {
-		if ( typeof propValue !== 'undefined' ) {
-			updateBoundAttibute( propValue, attrValue );
-		} else if ( placeholder ) {
-			/*
-			 * Placeholder fallback.
-			 * If the attribute is `src` or `href`,
-			 * a placeholder can't be used because it is not a valid url.
-			 * Adding this workaround until
-			 * attributes and metadata fields types are improved and include `url`.
-			 */
-			const htmlAttribute =
-				getBlockType( blockName ).attributes[ attrName ].attribute;
-
-			if ( htmlAttribute === 'src' || htmlAttribute === 'href' ) {
-				updateBoundAttibute( null );
-				return;
-			}
+	if ( typeof propValue !== 'undefined' ) {
+		updateBoundAttibute( propValue, attrValue );
+	} else if ( placeholder ) {
+		/*
+		 * Placeholder fallback.
+		 * If the attribute is `src` or `href`,
+		 * a placeholder can't be used because it is not a valid url.
+		 * Adding this workaround until
+		 * attributes and metadata fields types are improved and include `url`.
+		 */
+		const htmlAttribute =
+			getBlockType( blockName ).attributes[ attrName ].attribute;
 
-			updateBoundAttibute( placeholder );
+		if ( htmlAttribute === 'src' || htmlAttribute === 'href' ) {
+			updateBoundAttibute( null, attrValue );
```

Comment:
Unlike before, we pass the second parameter (`attrValue`) as well so that the function can bail [here](https://github.com/WordPress/gutenberg/pull/59443/files#diff-99d79bdba99fe4517413b0fd9503eef924e439c03d44504c1ad0606b727cebb6L109) if both params are the same.